metaflow/plugins/aws/aws_utils.py

```diff
@@ -4,6 +4,22 @@
 from metaflow.exception import MetaflowException
 
 
+def parse_s3_full_path(s3_uri):
+    from urllib.parse import urlparse
+
+    #  <scheme>://<netloc>/<path>;<params>?<query>#<fragment>
+    scheme, netloc, path, _, _, _ = urlparse(s3_uri)
+    assert scheme == "s3"
+    assert netloc is not None
+
+    bucket = netloc
+    path = path.lstrip("/").rstrip("/")
+    if path == "":
+        path = None
```

Comment:
we don't expect this to be `None`

Reply:
Yep, removing it makes sense for this use case.. I sound like a broken record at this point, but I just copied this wholesale from `parse_gcs_full_path`.